src/engine/camera.js

```diff
@@ -1,10 +1,17 @@
 import { PerspectiveCamera } from "three";
+import State from "./state";
 
-const Camera = new PerspectiveCamera(
-  75,
-  window.innerWidth / window.innerHeight,
-  0.1,
-  10000
-);
+class TrackingCamera extends PerspectiveCamera {
+  constructor() {
+    super(75, window.innerWidth / window.innerHeight, 0.1, 10000);
+  }
+
+  Update() {
+    if (!State.isXRSession) return;
+    this.matrixWorld.decompose(this.position, this.quaternion, this.scale);
```

Comment:
Moved the decompose to the Update loop which actually made the process even faster. ~0.008ms on my machine opposed to ~0.04ms when run on the XRReferenceFrame update loop.